underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -91,17 +91,24 @@ public SwiftUnderFileSystem(AlluxioURI uri, Configuration configuration) {
     }
     config.setAuthUrl(configuration.get(Constants.SWIFT_AUTH_URL_KEY));
     String authMethod = configuration.get(Constants.SWIFT_AUTH_METHOD_KEY);
-    if (authMethod != null && authMethod.equals("keystone")) {
-      config.setAuthenticationMethod(AuthenticationMethod.KEYSTONE);
+    if (authMethod != null) {
       config.setUsername(configuration.get(Constants.SWIFT_USER_KEY));
       config.setTenantName(configuration.get(Constants.SWIFT_TENANT_KEY));
-    } else {
-      config.setAuthenticationMethod(AuthenticationMethod.TEMPAUTH);
-      // tempauth requires authentication header to be of the form tenant:user.
-      // JOSS however generates header of the form user:tenant.
-      // To resolve this, we switch user with tenant
-      config.setTenantName(configuration.get(Constants.SWIFT_USER_KEY));
-      config.setUsername(configuration.get(Constants.SWIFT_TENANT_KEY));
+      switch (authMethod) {
+        case "keystone":
+          config.setAuthenticationMethod(AuthenticationMethod.KEYSTONE);
+          break;
+        case "swiftauth":
```

Comment:
nit: Can you please add a comment pointing to the docs of "swiftauth" ?

Seems like I can't find this in http://docs.openstack.org/developer/swift/overview_auth.html